docs/cue/reference/examples.cue

```diff
@@ -0,0 +1,80 @@
+package metadata
+
+#ConfigExample: {
+	title: !=""
+	example: !=""
+}
+
+config_examples: [#ConfigExample, ...#ConfigExample] & [
+	{
+		title: "Redacted Datadog Agent logs to Datadog"
+		example: #"""
+			[sources.datadog_agent]
+			type = "datadog_logs"
+			address = "localhost:80"
+
+			[transforms.remove_sensitive_user_info]
+			type = "remap"
+			inputs = ["datadog_agent"]
+			source = '''
+			  redact(., filters: ["us_social_security_number"])
+			'''
+
+			[sinks.datadog_backend]
+			type = "datadog_logs"
+			inputs = ["remove_sensitive_info"]
+			default_api_key = "${DATADOG_API_KEY}"
+			"""#
+	},
+	{
+		title: "Kafka topic to Elasticsearch"
+		example: #"""
+			[sources.kafka_in]
+			type = "kafka"
+			bootstrap_servers = "10.14.22.123:9092,10.14.23.332:9092"
+			group_id = "vector-logs"
+			message_key = "message"
+			topics = ["logs-*"]
+
+			[transforms.json_parse]
+			type = "remap"
+			source = '''
+			  . = merge!(parse_json!(.message))
+			'''
```

Comment:
```suggestion
			[transforms.json_parse]
			type = "remap"
			source = '''
			parsed, err = parse_json!(.message)
			if err != null {
			  log(err, level: "error")
			}
			  . = merge(parsed ?? {})
			'''
```

cc @jszwedko talked to @lucperkins about showing "best practices" in our examples - does this track with your expectations?

Reply:
@spencergilbert The `!` can be removed here, right?

Reply:
Yeah, I like showing the error handling as "best practice". You should remove the `!` there.

Reply:
And the last bit should be `. |= parsed ?? {}` or `. = merge(., parsed ?? {})`.

Reply:
Er, yea - I think a bad copy + paste

Reply:
> And the last bit should be `. |= parsed ?? {}` or `. = merge(., parsed ?? {})`.

LOL - I'm bad at things